backend/src/ml_space_lambda/utils/iam_manager.py

```diff
@@ -120,6 +120,23 @@ def __init__(self, iam_client=None, sts_client=None):
         self.default_notebook_role_policy_arns = []
         self.permissions_boundary_arn = env_variables[EnvVariable.PERMISSIONS_BOUNDARY_ARN]
 
+    def get_iam_role_arn(self, project_name: str, username: str) -> Optional[str]:
+        """
+        Get the ARN of an existing dynamic role for this (project_name, username) pair.
+
+        Args:
+            project_name (str): The project name
+            username (str): The username of the user
+
+        Return:
+            Optional[str]: The ARN of the existing role, otherwise None if no such role exists.
+        """
+
+        iam_role_name = self._generate_iam_role_name(username, project_name)
+
+        # Check if the IAM role exists already
+        return self._check_iam_role_exists(iam_role_name)
```

Comment:
Not on you but I had to look up what this does:`_check_iam_role_exists`

I was expecting it to return true/false based on the name

Reply:
I'm open to renaming it if you have something specific in mind that is a bit more descriptive…

- `_fetch_iam_role`
- `_get_iam_role` (maybe a little confusing with my new `get_iam_role_arn` maybe they're different enough since one is public and the "private"?)

Reply:
I think `_fetch_iam_role` works fine. Thanks for fixing!